internal/dag/httpproxy_processor.go

```diff
@@ -102,6 +103,28 @@ func (p *HTTPProxyProcessor) Run(dag *DAG, source *KubernetesCache) {
 	}
 }
 
+// Credits to https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L95
+// Had to rewrite it here as we don't need FieldPath
+func (p *HTTPProxyProcessor) isValidFQDN(name string) []string {
```

Comment:
This looks great! Unfortunately, because of Kube things, we actually need to allow a single DNS label as an option as well. I'm sorry not to have made that clear earlier. So, the check about having at least 2 segments can't be added, as it's an API change.

So, this function should check:
Is this a valid DNS subdomain (using the validation check)?
Are all the labels valid (one is a valid number of labels)?
If not, it's not valid.

For this new function, I'd also like to see unit tests that check each if statement adds the correct errors.